### ISSUE TITLE
Fusee Suite not starting

### ISSUE THREAD

**Issue by ghost** [mattytrog/FUSEE_SUITE#5]
TL:DR: Trinket starts switch in RCM but won't inject the payload

Hey,
I had my trinket working without any problems for over a year. But because of some random issues with my joycons I did a fresh switch setup. I also updated my trinket. But now it only lights up the red power led when starting the switch. The multicolour LED stays off all the time. My switch ends up in RCM mode but the payload isnt pushed (I dont see the SD card missing/ Fusee_Suite bootscreen). I also reflashed the trinket multiple times by pressing the reset button when the switch is booted. I can upload both the first and the second uf2 without any issues. Yet I am unable to get the trinket working.
Am I missing something?
Or might it be that my trinket has gone defective?

Would appreciate any help or pointers what to try

**Comment by mattytrog** [comment]
Hello!

Right... I'm guessing you flashed the new part 1?

Sounds like you only flashed part 1 to me. Do this:

Either hold reset button in(push back cover in on switch) or let it trigger
RCM mode, then push a payload and let the console boot normally...

Once booted, let screen go off and you should see a "FUSEE" drive pop up.
It's this we are aiming for. You might need to double press the back
cover(to enter uf2 mode).

Once you have done this, drag and drop part 2.

Let me know how you get on

**Comment by mattytrog** [comment]
I must add that if you are flashing a new part 1, you need to flash a new
part 2 as the flash is overwritten.

You can however flash a new part 2 without flashing part 1 again.

Cheers
Matty

**Comment by ghost** [comment]
Thank you for your fast response.
Unfortunately I did flash part 1 first and after the drive reappeared I flashed part 2. 
INFO_UF2 inside the "FUSEE" drive also matches the part 1
`UF2 Bootloader c40fc1c SFHWRO
Model: Trinket M0
Board-ID: SAMD21E18A-Trinket-v0
`
But still only the power LED briefly flashes and the switch ends up in RCM.

Should I try a specific old version? Unfortunately I am unaware which Version I previously used

**Comment by mattytrog** [comment]
You have the Fusee drive. Brilliant. Now you need to choose a part 2.

If you have already drag and dropped a part 2, all you need to do is power
the switch down.

The Fusee drive opens up out of courtesy in case you want to flash a
"settings.uf2" file, upon a successful part 2 flash.

Kind regards
Matty

**Comment by mattytrog** [comment]
Also... You have unplugged from pc usb haven't you?

It cannot push a payload while connected to pc, as the pc "kidnaps" the
port. Disconnect and try booting

**Comment by mattytrog** [comment]
Also also... Have a read on the main page about FailsafeRCM. If you have USB strap fitted, that is.

**Comment by ghost** [comment]
Yeah I did flash part 2 and disconnected the Switch before rebooting. But I just realized that I didnt resolder the optional USB Disconnect strap. Will connect it and report back

**Comment by mattytrog** [comment]
It isn't mandatory to have it fitted.

Your checklist should thus be:

- All required straps connected(should be as it was working) - OK
- Part 1 flashed, turning Trinket from SWITCHBOOT to FUSEE - OK
- Part 2 flashed - Will open up FUSEE on first boot in case you want a SETTINGS.UF2 on there. If not just power down.
- Unplug from USB, try booting.
- If no boot, eject SD card and try booting again. Does it take you to Fusee main page?
- Also, you dont need to keep VOL+ held. Just hold until it tells you to let go, if you want the menu, or press momentarily if CFW booting.

**Comment by ghost** [comment]
I flashed part 1
Drive reopened and I flashed part 2
Drive reopened and I disconnected
Turned off the switch
Powered on and ended up in RCM.
I dont see the Fusee main page. Even if I remove the SD Card. And the multicolour LED stays off all the time.

(I then connect it to my pc and inject the payload via TegraRCM to exit RCM and start my switch)

**Comment by mattytrog** [comment]
OK... If you boot into the Switch console, you double-press the back cover, does it go into UF2(Fusee) mode?

Or hold VOL+ down for about 20 seconds... Any LEDS flashing after 4 seconds or so?

**Comment by mattytrog** [comment]
The only time I have seen something like this, is when there has been an issue with the USB D+ and D- wires (connected wrong way round). Though Fusee opened - so this cannot be the case.

Part 1 is working, because you are entering RCM. If part 1 wasn't working, RCM wouldnt activate.
You don't have autoRCM installed do you from Hekate? This isn't required.

**Comment by mattytrog** [comment]
If you can boot your Switch (by however means - push payload or whatever) and you can access Fusee drive, I'd just drag and drop part 2 again.

**Comment by ghost** [comment]
Yes when i double press it see the fusee drive.
 I just held down volume up. After around 5 seconds it starts a few short flashes followed by a longer flash (All in green). I held it for a few seconds and the pattern repeated itself. But afterwards the power LED stayed on.
 
 After a reboot it isnt still working, but I just realized that both the Power LED and the multiclour LED do a short red flash.
 
 Auto RCM is not activated. I also checked and the switch wont go into RCM when the trinket is disconnected. Also after powering down the switch it is not in RCM. Only once i press the power button it enters RCM

**Comment by ghost** [comment]
I also tried just reuploading part 2 but the issue persists.
And ive redone the soldering at least one time so I am quite sure its not a connection issue (especially because all the Straps for RCM work. And as I can access the FUSEE drive the Data straps should also work

**Comment by mattytrog** [comment]
Goes into RCM... Brilliant.

A short red flash. Brilliant.

Green flashes... Brilliant.

That means part 2 is successfully flashed.
The only problem remaining is why the Switch is not getting the payload. The RED little flash signals "push failure". Its green if successful

If you are in RCM, are disconnected from PC, are seeing the red flash... And seeing green if you hold VOL, your Trinket is working normally.

That leaves the USB D+ and D- wires or pads. Nothing else it can be. Are they good solder joints? Can you get back into Fusee drive if booted to Switch lock screen(then putting screen to sleep?)

**Comment by ghost** [comment]
Yes I can get back into Fusee drive if booted to the lockscreen and putting the screen to sleep. This worked every time without any issue (Did at least 10 reflashes)

Solder joints also looked fine to me. Data Straps are approx 2 cm long. I already resoldered them before but I can give it another shot. (Still I am confused why they would work when the Switch is booted but dont work when the trinket is trying to inject the payload)

**Comment by mattytrog** [comment]
Ok... Ill investigate this further. In meantime flash an old part 2 if you
like. See if any boot. They should all work

**Comment by mattytrog** [comment]
I'm going to commit a possible fix. I think it might be injecting too fast
for certain units. I'll be home in next half hour and will do a testing
trinket build this evening.

**Comment by ghost** [comment]
I just flashed the `Part_1_TRINKET_PERMA_CFW_Latest.uf2` followed by `TRINKET.UF2` in your `MODCHIP_LATEST` Repo. Now I get the LED flashing when trying to boot that I was used to. But I still end up in RCM. Because even these old files arent working with my switch it seems to me that it is indeed a problem with my switch/soldering and not the code. (I also tried it without a SD Card)

On another note, with this old version if I press vol up after my switch gets stuck in RCM the multiclolour LED blinks red. Am I right to assume that the trinket is trying to reinject the payload but keeps failing? That would mean that it isnt a timing issue either

**Comment by mattytrog** [comment]
Yes. Red = failure. The failure is either finding RCM(99.999999% likely) or actually pushing the payload. I have changed the RCM seek time to 1 second in this build. Can you try this please?
[TRINKET.zip](https://github.com/mattytrog/FUSEE_SUITE/files/8995751/TRINKET.zip)

**Comment by ghost** [comment]
Which part1 should I use? Just the latest one ?

**Comment by mattytrog** [comment]
yes please

**Comment by ghost** [comment]
Unfortunately no luck. Still end up in RCM. But now the multicolour LED didnt flash red (just the power LED)

**Comment by mattytrog** [comment]
OK... Power the unit off and try once more, unless you have tried this. Can you try another version for me?

**Comment by ghost** [comment]
Already tried it twice but I can try it another time. Also happy to test any other files
Edit: Tried it two more times. Same behaviour. Now only the Power LED blinks once

**Comment by mattytrog** [comment]
Thank-you. I cannot replicate the problem you see... OK Test number 2
[TRINKET_TEST2.zip] << Broken

**Comment by ghost** [comment]
I dont know if its relevant but after I flashed the UF2 from `Trinket.zip` the FUSEE drive didnt show up again autmatically. Gonna try the new file now

**Comment by mattytrog** [comment]
Hang on... I discovered problem... Test 3 coming. Oops..
[TRINKET_TEST3.zip](https://github.com/mattytrog/FUSEE_SUITE/files/8995836/TRINKET_TEST3.zip)
.

**Comment by ghost** [comment]
Okay v2 was the same as v1(from my perspective)
With v3 at least the multicolour LED is flashing again. Flashing at the same time as the power LED. But I still end up in RCM (Tested twice)

**Comment by mattytrog** [comment]
Is it going green at all when you first boot up?

**Comment by ghost** [comment]
When I press the power button? Then it is just flashing red at the same time as the power LED. Or should I retry holding vol up with the switch booted?

**Comment by mattytrog** [comment]
Yes please. Also, if you can push a payload via PC (tegraRCMsmash or whatever)... Can you push this and tell me if it works?
[testpayload.zip](https://github.com/mattytrog/FUSEE_SUITE/files/8995868/testpayload.zip)

**Comment by ghost** [comment]
I tried holding vol up 2 times. The first time it did the fast green flashes followed by a single longer green flash. There also was a short red flash after.
The second time I tried it I got the same green flashes followed by the same pattern in red

**Comment by ghost** [comment]
And now it just booted into the stock fw. Did I just disable Chip_RCM?

**Comment by mattytrog** [comment]
Yes. Thats what the green hold vol+ flashes do

**Comment by mattytrog** [comment]
just hold vol+ again until the first flash, let go, to trigger ChipRCM

**Comment by mattytrog** [comment]
Are you able to push the payload I uploaded(testpayload) above?

**Comment by mattytrog** [comment]
When you disabled ChipRCM, you put your console into dual-boot mode. Its a toggle

**Comment by ghost** [comment]
I am unable to enable chiprcm
When I stop holding vol up whilst it is flashing green it still flashes red afterwards. When I keep holding it it will just continue its green pattern, followed by the red pattern whenever i let go

**Comment by ghost** [comment]
But whatever gonna try to inject the payload first

**Comment by ghost** [comment]
Injecting the payload worked. Saw the red Fusee_Suite screen and ended up in hekate

**Comment by mattytrog** [comment]
OK... In OFW mode, you should be able to get into the FUSEE drive... If you can, flash this... This will re-enable ChipRCM
[SETTINGS_B_JVU_B1_2_LP.zip](https://github.com/mattytrog/FUSEE_SUITE/files/8995929/SETTINGS_B_JVU_B1_2_LP.zip)

**Comment by mattytrog** [comment]
And stand by. I'm building another test. I've never ever seen a problem like this before...

**Comment by ghost** [comment]
ChipRcm is now reenabled. Actually I am just highly confused why it isnt working. Tbh I doubt it is a software bug as it worked before. But I am also confused why rcm, power and file transfer works but the trinket cant push the payload. I just cant wrap my head around it

**Comment by mattytrog** [comment]
Me neither. Try this. I have removed the vendor ID check in case it has a different vendor ID

[TEST_4_TRINKET.zip](https://github.com/mattytrog/FUSEE_SUITE/files/8995959/TEST_4_TRINKET.zip)

**Comment by mattytrog** [comment]
You are getting the Fusee drive... RCM is being triggered... The switch is unpatched. The LED is blinking red (one blink) - that means it isn't seeing RCM from the Trinket.

**Comment by ghost** [comment]
Still the same :(
I guess I am just out of luck

**Comment by mattytrog** [comment]
OK... Nuclear option. I'm going to try a build without the RCM check... Stand by...

**Comment by mattytrog** [comment]
Try this...
[TEST_5_TRINKET.zip](https://github.com/mattytrog/FUSEE_SUITE/files/8996000/TEST_5_TRINKET.zip)

**Comment by ghost** [comment]
Still no difference

**Comment by mattytrog** [comment]
Is the LED going green now? the single blink?

**Comment by ghost** [comment]
Power LED and Multi LED shortly flash red. And then I am in RCM

**Comment by ghost** [comment]
You mean once again when I hold vol up ? Or is the Multi LED supposed to light up green on boot?

**Comment by mattytrog** [comment]
OK... Stand by... Sorry about this - hope you are OK for time.

**Comment by mattytrog** [comment]
OK... Not expecting this to work... But lets try.
[TRINKET_TEST6.zip](https://github.com/mattytrog/FUSEE_SUITE/files/8996018/TRINKET_TEST6.zip)

**Comment by mattytrog** [comment]
No dont hold VOL up. On successful payload push / RCM found, that single little (Red in your case) LED flash, should be green

**Comment by ghost** [comment]
Still the red flashes with both LEDs. Via what straps does the Trinket tell that the switch is in RCM?

**Comment by mattytrog** [comment]
Joycon and volume

**Comment by ghost** [comment]
Arent that the straps that the trinket uses to get the switch into RCM?

**Comment by mattytrog** [comment]
Ahhh... Joycon and volume trigger RCM... Then the Trinket looks for the Switch in RCM mode (just as though it is plugged in to the USB port). So the Switch says to the Trinket "Hey, I'm in RCM" and it does this via USB.

**Comment by mattytrog** [comment]
OK... Daft question... How have you got the Trinket fitted? As my diagram? I'm wondering if the voltage is sagging because there have been substantial changes to part 1.

**Comment by ghost** [comment]
I am using the alt points on the back for 3.3, joycon and vol+. And as I mentioned I currently havent connected the USB disconnect strap. But I can also resolder 3.3 to the front if that might help. Or can I power the trinket externally for testing ?

**Comment by ghost** [comment]
I can also try to use a multimeter to check whether vcc is collapsing but I guess it will be too quick for me to notice

**Comment by mattytrog** [comment]
OK... Potential things to check... Check the 3v3 point. Check its not too long. You could also remove the 5 pin regulator in case that is dragging the V+ down(there will be a certain amount of resistance between the outputs and ground. I'm wondering if this regulator is being greedy with the 3v3. If you do remove this, you will no longer be able to power from USB...

As an alternative, disconnect the 3v3 and try connecting Trinket to battery 5v(so battery + to Trinket 5v(or USB pad i think it is, next to pin 0)

**Comment by mattytrog** [comment]
If the Trinket then pushes the payload with the 5v USB connection soldered, (remember it will be permanent on) then it is indeed the 3v3 sagging. You will probably need to push reset on the Trinket once you are in RCM because it will already be powered.

**Comment by ghost** [comment]
Ill try powering using 5v first

**Comment by ghost** [comment]
Is there any known 5V point but the battery connector ?

**Comment by mattytrog** [comment]
Yes. Just next to the battery connector should be.

**Comment by ghost** [comment]
I am getting ~0V next to the battery

**Comment by mattytrog** [comment]
There is definitely a 5v point there somewhere. I cannot remember where exactly it is.

**Comment by ghost** [comment]
Cant seem to find it tho. But I could remove the regulator and see whether that helps

**Comment by mattytrog** [comment]
Before you do that, try some really early part 1 and part 2 please. See if they working.

**Comment by mattytrog** [comment]
Try version 3 or V6, with the old part 1

**Comment by ghost** [comment]
Tried v3 with old part 1. A few yellow flashes followed by a longer and a shorter red. Still ended up in RCM

**Comment by ghost** [comment]
I guess Ill order another Trinket and resolder everything to see whether that fixed it

**Comment by mattytrog** [comment]
Ok... This is beyond weird. Is there any way you can do a continuity test between your d+ /d- on trinket and end of usb cable( where it plug into pc might be easiest). So plug usb lead in to switch and continuity test the middle data pins(pc end) to the points on the trinket?

**Comment by ghost** [comment]
Just to check whether I understood you correctly. Ill plug in a USB cable and check for continuity from d+ to d+ and afterwards from d- to d- ?

**Comment by ghost** [comment]
Everything seems to be fine. Have continuty from one of the middle pins to d+ and continuity from the other to D-

**Comment by mattytrog** [comment]
Yes. So from the trinket(d+) where the usb port used to be, to end of the cable. Same with d-

**Comment by ghost** [comment]
Yep that works

**Comment by mattytrog** [comment]
Honestly... I'm at a loss. The issue is the trinket not seeing the switch is in RCM. So if no versions are working, there is an issue elsewhere. Usb wires too long, power sagging. Could the battery even be very unsyncd... Crap inside the usb port giving a very high impedance between d+/d-. Tomorrow I will try to replicate the issue some more. You could try removing regulator...

**Comment by ghost** [comment]
No worries. I highly appreciate your effort. I just ordered another trinket. Once it is arrived I will give it another shot and comment my findings. Maybe it will work. But otherwise ill take the loss. At least it did work for a long time (and the one I installed for a friend is still working fine)
Awesome project btw

**Comment by mattytrog** [comment]
Very kind of you to say. I just wanted to rule out software issues(which I think we have at this point). Try another trinket and check the usb port for debris. What's puzzling though is it was working, but now it isn't. With any version. There is a filter just above the usb pads(on some units) on the switch motherboard. Tiny little black ic. This could be pulling the d lines down. The pc will have stronger ones and it is before the tegra, so Fusee drive will still show. It is safe to remove this

**Comment by ghost** [comment]
I'll first wait for the new trinket and try other stuff afterwards. The behaviour is so weird that I am hoping my trinket has gone crazy

**Comment by mattytrog** [comment]
Yeah it is weird. Definitely a high impedance somewhere. That filter I mentioned is only fitted in some markets for whatever reason. It's a new one for me and I've been doing this sh.t for nearly 5 years. I'll investigate more tomorrow. I'll leave this on. Actually I better close it because it's not the software. I'll reopen if needed

**Comment by ghost** [comment]
FYI: Trinket arrived today. Flashed and soldered the new trinket: Same behaviour. Resoldered d+ and D-: Same behaviour. Resoldered every connection and inspected every solder point for bridges: Same behaviour. I guess I'll ditch the Trinket and get a RCM Loader or OTG USB Cable

**Comment by ghost** [comment]
With the trinket installed I can inject from my PC. But I cant inject from my Smartphone (USB C to USB C). I am also unable to change the USB mode on my smartphone to  Host (USB controlled via smartphone). But once I unsoldered VCC I can inject with my phone. Any reason why this would be the case? Thought about using the Trinket just for RCM

**Comment by mattytrog** [comment]
I'm wondering if that filter above the front usb pads is playing up. If you
have the facilities, check the impedance of the usb d lines to gnd. Then
remove filter and recheck. You cannot inject from a phone... Can you inject
from a phone with trinket disconnected?

**Comment by ghost** [comment]
Impedance or resistance?

With the trinket disconnected(just 3.3V disconnected but d+ and d- still connected) I can inject with my phone. When the trinket is connected to VCC I cannot inject. But maybe that's because the trinket tries to inject once USB power is connected, doesn't it? Just curious why it works on the PC but with my phone. But maybe that's just android behaving

**Comment by mattytrog** [comment]
I wanted to know what the continuity was like in the usb traces on the
switch. It goes USBC>test pads(trinket pads)>filter> then through some vias
to the underside.

The current/voltage levels are higher coming from a pc usb port and could
still talk to the switch in RCM and the Fusee drive. These are before that
filter. It seems like the switch cannot "hear" the trinket because
something is making the data traces "deaf". I'm thinking that filter.

**Comment by mattytrog** [comment]
Look at these points. The big ic appears to be some kind of filter. It's
only fitted to some markets. The little one is a ferrite choke. Take some
readings and maybe remove the big one. Ensure the choke isn't straying to
gnd

**Comment by mattytrog** [comment]
Also... Try going straight to the alt usbs on the back from trinket...

**Comment by mattytrog** [comment]
I don't think an external battery based injector will help you. Unless it's
a laptop xx
